File: src/main/java/br/com/rotaractsorocabaleste/rotasales/user/UserService.java
package br.com.rotaractsorocabaleste.rotasales.user;

import org.springframework.security.core.GrantedAuthority;

import java.util.Collection;
import java.util.Optional;

public interface UserService {

    User create(final User user);

    Optional<User> findByUsername(final String username);

    User findLoggedInUser();

    void changePassword(final ChangePasswordDTO changePasswordDTO);

    Collection<GrantedAuthority> findAuthoritiesByUsername(final String username);

}

File: src/main/kotlin/br/com/rotaractsorocabaleste/rotasales/core/service/UserService.kt
package br.com.rotaractsorocabaleste.rotasales.core.service

import br.com.rotaractsorocabaleste.rotasales.core.entity.User

interface UserService {
    fun save(user: User): User
    fun findByUsername(username: String): List<User>
    fun getLoggedInUser(): User
}